### PR TITLE
chore(jql-parser): improve parser implementation based on @epage 's feedback

### PR DESCRIPTION
**Pull request by yamafaktory**
This is follow-up PR based on the feedback from #245.

### CODE REVIEW

File: crates/jql-parser/src/combinators.rs [hunk 1]
```diff
@@ -189,27 +200,27 @@ pub(crate) fn parse_lens<'a>(
...
+        separated(1.., parse_lens, trim(COMMA)),
+        trim(CURLY_BRACKET_CLOSE),
+    )
     .parse_next(input)
 }
 
 /// A combinator which parses a flatten operator.
 pub(crate) fn parse_flatten_operator<'a>(input: &mut &'a str) -> PResult<&'a str> {
-    trim(FLATTEN).parse_next(input)
+    tag(FLATTEN).parse_next(input)
```
Comment (epage) [inline crates/jql-parser/src/combinators.rs:213]: `tag` shouldn't be needed.  Did you get an error?

Comment (yamafaktory) [reply crates/jql-parser/src/combinators.rs:213]: I introduced it since I got `cannot borrow immutable static item `FLATTEN` as mutable`.

Comment (epage) [reply crates/jql-parser/src/combinators.rs:213]: Forgot about that; thanks

Comment (yamafaktory) [reply crates/jql-parser/src/combinators.rs:213]: Thanks a ton for your feedback @epage. Resolving the conversation now and moving on with the changes.

**Pull request merged**